### PR TITLE
BOLT11: Implement change in r field

### PR DESCRIPTION
**Pull request by sstone**

### CODE REVIEW

**Comment by btcontract** [comment]
umm it's already implemented in https://github.com/ACINQ/eclair/pull/273, along with multiple simplifications: thanks to base+proportional `buildExtra` method is not needed at all now and `Hop` with `ExtraHop` don't need to inherit a common `PaymentHop` trait.

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala [hunk 1]
```diff
@@ -217,15 +217,16 @@ object PaymentRequest {
     *
     * @param nodeId          node id
     * @param shortChannelId  channel id
-    * @param fee             node fee
+    * @param fee_base_msat   node fixed fee
+    * @param fee_proportional_millionths  node proportional fee
     * @param cltvExpiryDelta node cltv expiry delta
     */
-  case class ExtraHop(nodeId: PublicKey, shortChannelId: Long, fee: Long, cltvExpiryDelta: Int) extends PaymentHop {
+  case class ExtraHop(nodeId: PublicKey, shortChannelId: Long, fee_base_msat: Long, fee_proportional_millionths: Long, cltvExpiryDelta: Int) extends PaymentHop {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala]: Variable names should be camel case

**Comment by sstone** [comment]
Yes sorry I could have asked you to split your PR instead. Right now I'd rather have a minimal PR that implements the latest BOLT11 changes, and a separate one that implements how we use the extra routing info embedded in a payment request.

File: eclair-core/src/test/scala/fr/acinq/eclair/payment/PaymentRequestSpec.scala [hunk 2]
```diff
@@ -106,7 +106,10 @@ class PaymentRequestSpec extends FunSuite {
     assert(pr.nodeId == PublicKey(BinaryData("03e7156ae33b0a208d0744199163177e909e80176e55d97a2f221ede0f934dd9ad")))
     assert(pr.description == Right(Crypto.sha256("One piece of chocolate cake, one icecream cone, one pickle, one slice of swiss cheese, one slice of salami, one lollypop, one piece of cherry pie, one sausage, one cupcake, and one slice of watermelon".getBytes)))
     assert(pr.fallbackAddress === Some("1RustyRX2oai4EYYDpQGWvEL62BBGqN9T"))
-    assert(pr.routingInfo() === List(RoutingInfoTag(List(ExtraHop(PublicKey("029e03a901b85534ff1e92c43c74431f7ce72046060fcf7a95c37e148f78c77255"), 72623859790382856L, 20, 3), ExtraHop(PublicKey("039e03a901b85534ff1e92c43c74431f7ce72046060fcf7a95c37e148f78c77255"), 217304205466536202L, 30, 4)))))
+    assert(pr.routingInfo() === List(RoutingInfoTag(List(
+      ExtraHop(PublicKey("029e03a901b85534ff1e92c43c74431f7ce72046060fcf7a95c37e148f78c77255"), 72623859790382856L, 1, 20, 3),
```
Comment (pm47) [inline eclair-core/src/test/scala/fr/acinq/eclair/payment/PaymentRequestSpec.scala]: To make translation from test vectors more obvious:

`72623859790382856L` -> `0x0102030405060708L`
`217304205466536202L` -> `0x030405060708090aL`

**Pull request merged**